File: src/main/java/com/wisegas/grapp/itemmanagement/domain/value/GrappItemCodeType.java
package com.wisegas.grapp.itemmanagement.domain.value;

public enum GrappItemCodeType {
   GTIN("%013d"),
   UPC("%012d"),
   NACS("%02d%02d%02d"),
   MANUAL("%s"),
   RANDOM("%s");

   private final String valueFormat;

   public String getValueFormat() {
      return valueFormat;
   }

   GrappItemCodeType(String valueFormat) {
      this.valueFormat = valueFormat;
   }
}

File: src/main/webapp/app/items/GrappItemCodeType.js
(function() {
   "use strict";

   angular.module("Grapp")
      .constant("GrappItemCodeType", createEnum());

   function createEnum() {
      return {
         GTIN: new GrappItemCodeType("GTIN"),
         UPC: new GrappItemCodeType("UPC"),
         NACS: new GrappItemCodeType("NACS"),
         MANUAL: new GrappItemCodeType("MANUAL")
      };
   }

   function GrappItemCodeType(displayString) {
      this.displayString = displayString;
   }
})();